packages/core/src/state-management/job-queue.ts

```diff
@@ -1,10 +1,19 @@
-import { default as PgBoss } from 'pg-boss'
+import { default as PgBoss, type SendOptions } from 'pg-boss'
 import Pg from 'pg'
 import { fromEvent, firstValueFrom } from 'rxjs'
 
+export type Job = {
+  name: string
+  data: object
```

Comment:
Any way we can refine the data needed by a job please?
If it differs between jobs, maybe we can make the `Job` type generic with a type parameter for the `data`? Or if the `data` is based on the job's `name`, can we define the mapping to benefit from the type validation?